src/test/java/org/gridsuite/ds/server/service/client/timeseries/TimeSeriesClientTest.java

```diff
@@ -7,108 +7,174 @@
+                ));
+
+        // test service
+        UUID timeSeriesUuid = timeSeriesClient.sendTimeSeries(timeSeries).getId();
+
+        // check result
+        assertEquals(TIME_SERIES_UUID, timeSeriesUuid);
+
     }
 
     @Test
-    public void testSendTimeSeries() {
-        Map<String, TimeSeries> curves = new HashMap<>();
-        TimeSeriesIndex index = new IrregularTimeSeriesIndex(new long[]{32, 64, 128, 256});
-        curves.put("NETWORK__BUS____2-BUS____5-1_AC_iSide2", TimeSeries.createDouble("NETWORK__BUS____2-BUS____5-1_AC_iSide2", index, 333.847331, 333.847321, 333.847300, 333.847259));
-        curves.put("NETWORK__BUS____1_TN_Upu_value", TimeSeries.createDouble("NETWORK__BUS____1_TN_Upu_value", index, 1.059970, 1.059970, 1.059970, 1.059970));
-        List<TimeSeries> timeSeries = new ArrayList<>(curves.values());
-        UUID timeSeriesUuid = timeSeriesClient.sendTimeSeries(timeSeries).block().getId();
-        assertEquals(TIME_SERIES_UUID, Optional.of(timeSeriesUuid).orElseThrow().toString());
+    public void testSendTimeSeriesGivenException() {
+        // prepare time series
+        List<TimeSeries<?, ?>> timeSeries = createTimeSeriesList();
+
+        // mock response for test case POST with url - /timeseries-group
+        String baseUrl = getEndpointUrl();
+
+        wireMockServer.stubFor(WireMock.post(WireMock.urlMatching(baseUrl + ".*"))
+                .willReturn(WireMock.serverError()
+                        .withBody(ERROR_MESSAGE_JSON)));
+
+        // test service
+        DynamicSimulationException dynamicSimulationException = catchThrowableOfType(
+                () -> timeSeriesClient.sendTimeSeries(timeSeries),
+                DynamicSimulationException.class);
+
+        // check result
+        assertThat(dynamicSimulationException.getType())
+                .isEqualTo(CREATE_TIME_SERIES_ERROR);
+        assertThat(dynamicSimulationException.getMessage())
+                .isEqualTo(ERROR_MESSAGE);
+
+    }
+
+    @Test
+    public void testSendTimeSeriesGivenEmpty() {
+        // prepare time series
+        List<TimeSeries<?, ?>> timeSeries = new ArrayList<>();
+
+        // test service
+        TimeSeriesGroupInfos timeSeriesGroupInfos = timeSeriesClient.sendTimeSeries(timeSeries);
+
+        // check result
+        assertEquals(null, timeSeriesGroupInfos);
 
     }
 
     @Test
     public void testDeleteTimeSeriesGroup() {
-        timeSeriesClient.deleteTimeSeriesGroup(UUID.fromString(TIME_SERIES_UUID)).block();
+
+        // mock response for test case DELETE with url - /timeseries-group
+        String baseUrl = getEndpointUrl();
+        wireMockServer.stubFor(WireMock.delete(WireMock.urlMatching(baseUrl + ".*"))
+                .willReturn(WireMock.ok()));
+
+        // test service
+        timeSeriesClient.deleteTimeSeriesGroup(TIME_LINE_UUID);
+
+        // check result
+        assertTrue(true);
```

Comment:
funny assertion, does sonar force you to make an assertion in a test ?

Reply:
YES, just satisfy SONAR! but now I use assertDoesNotThrow when testing a method which does not except any exception..

DONE in https://github.com/gridsuite/dynamic-simulation-server/commit/2c739715790cec6643c39f58027d1c2531dcca1b